### PR TITLE
Allow http proxy on bootstrap

### PR DESCRIPTION
**Pull request by feltra**

### CODE REVIEW

**Comment by bastelfreak** [comment]
Hi @feltra, thanks for the PR. Can you please rebase against our latest master branch? There are a bunch of commits here that should not appear.

File: manifests/pip/bootstrap.pp [hunk 1]
```diff
@@ -21,24 +22,26 @@
     }
     if $version == 'pip3' {
       exec { 'bootstrap pip3':
-        command => '/usr/bin/curl https://bootstrap.pypa.io/get-pip.py | python3',
-        unless  => 'which pip3',
-        path    => $python::params::pip_lookup_path,
-        require => Package['python3'],
+        command     => '/usr/bin/curl https://bootstrap.pypa.io/get-pip.py | python3',
+        environment => [ "HTTP_PROXY=${http_proxy}", "HTTPS_PROXY=${http_proxy}", "http_proxy=${http_proxy}", "https_proxy=${http_proxy}" ],
```
Comment (bastelfreak) [inline manifests/pip/bootstrap.pp]: why set it as uppercase and lowercase environment variables?

File: manifests/pip/bootstrap.pp [hunk 2]
```diff
@@ -11,6 +11,7 @@
 class python::pip::bootstrap (
   Enum['pip', 'pip3'] $version            = 'pip',
   Variant[Boolean, String] $manage_python = false,
+  String $http_proxy                      = '',
```
Comment (bastelfreak) [inline manifests/pip/bootstrap.pp]: please default to undef and enforce the minimal string length:
```suggestion
  String[1] $http_proxy                   = undef,
```

File: manifests/pip/bootstrap.pp (hunk 1 above)
Comment (feltra) [reply manifests/pip/bootstrap.pp]: AIX curl needed it to be lower case.

**Comment by bastelfreak** [comment]
can you please have a look at the failing travis jobs and rebase against our latest master branch and squash the commits?

**Comment by pillarsdotnet** [comment]
@bastelfreak Looks like tests are passing, and you can squash commits when merging.

Is anything else preventing this PR from being merged?

**Comment by bastelfreak** [comment]
Hi. This branch doesn't cleanly rebase on our current master. A squash via the web UI triggers a merge, even if CI would then fail. Please do a rebase on the command line and forcepush to this branch / resubmit it as a new PR.

**Comment by pillarsdotnet** [comment]
@bastelfreak I cannot push to this branch but I submitted #488 which is a merged version.

**Pull request closed without merging**